### PR TITLE
RFC: Squeeze and Excitation

### PR DESCRIPTION
**Pull request by sethtroisi**
A couple of things I'm worried about

1. I'm adding the wrong original value
2. I've placed the relu's at the wrong points
3. Should I place SE around initial_block?
4. Should SE be placed around the last block?
5. should I bother with averagepool or just call "reduce_mean(axis=[1,2])"?

Unrelated to this PR should we test NCHW? it was a appearently lot faster for Lc0 when using cudnn

Issues tracker: #683

### CODE REVIEW

File: dual_net.py [hunk 1]
```diff
@@ -375,19 +388,67 @@ def model_inference_fn(features, training, params):
...
+        int_layer1 = my_batchn(my_conv2d(inputs))
+        initial_output = tf.nn.relu(int_layer1)
+        int_layer2 = my_batchn(my_conv2d(initial_output))
+        residual = int_layer2
+
+        channels = params['conv_width']
+        ratio = FLAGS.SE_ratio
+
+        global_pool1 = my_global_avgpool2d(residual)
+        fc1 = tf.layers.dense(global_pool1, units=channels / ratio)
```
Comment (brilee) [inline dual_net.py]: should this be integer division (channels // ratio)

File: tests/test_flags [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # One line per flag
 # For example --someflag=value
---conv_width=1
+--conv_width=5
```
[the remark this answers is not in the document]
Comment (sethtroisi) [reply tests/test_flags]: So it's just hard to tell that channel is last when the weights are shape [18,1,1,1]

**Comment by sethtroisi** [comment]
A couple of things I'm worried about

1. I'm adding the wrong original value
2. I've placed the relu's at the wrong points
3. Should I place SE around initial_block?
4. Should SE be placed around the last block?
5. should I bother with averagepool or just call "reduce_mean(axis=[1,2])"?

Unrelated to this PR should we test NCHW? it was a appearently lot faster for Lc0 when using cudnn

File: dual_net.py [hunk 3]
```diff
@@ -375,19 +388,67 @@ def model_inference_fn(features, training, params):
...
+
+        int_layer1 = my_batchn(my_conv2d(inputs))
+        initial_output = tf.nn.relu(int_layer1)
+        int_layer2 = my_batchn(my_conv2d(initial_output))
+        residual = int_layer2
+
+        channels = params['conv_width']
+        ratio = FLAGS.SE_ratio
+
+        global_pool1 = my_global_avgpool2d(residual)
```
[the remark this answers is not in the document]
Comment (brilee) [reply dual_net.py]: but I guess batchnorm introduces inter-example dependency where a given example might be more positive/negative than the batch average? And I guess batchnorm is applied to each activation independently, so it should be okay-ish.

**Comment by brilee** [comment]
Took another look and it still looks good to me.

To answer questions
1. I'm adding the wrong original value - the bias term addition looks mildly weird to me, but then again, it's mildly weird that our original resnet implementation will take a previously relu'd output (so all entries >= 0), then add a batchnorm'd residual (so half of entries will be < 0, on average), then relu it again (immediately discarding a quarter of the residuals that happened to be added to 0 entries).
2. I've placed the relu's at the wrong points - see above. relus for SE part seem correct to me at least according to the paper.
3. Should I place SE around initial_block? Sure, this seems like a good idea. (The paper suggests that later SE blocks are less important, at least in imagenet applications where I guess you can have things like gross color inbalances.)
4. Should SE be placed around the last block? Sure, why not?
5. should I bother with averagepool or just call "reduce_mean(axis=[1,2])"? Don't think it matters other than as a performance hack.

Unrelated to this PR should we test NCHW? it was a appearently lot faster for Lc0 when using cudnn - We should probably optimize for whatever's faster on the TPU.

File: dual_net.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sethtroisi) [reply dual_net.py]: I don't have an explanation, I just coded it up. We can maybe explore these questions later.

**Comment by amj** [comment]
Did you mean to force-push?  Otherwise, yes LGTM
